core/src/main/scala/cats/syntax/either.scala

```diff
@@ -306,6 +308,14 @@ final class RightOps[A, B](val right: Right[A, B]) extends AnyVal {
   def leftCast[C]: Either[C, B] = right.asInstanceOf[Either[C, B]]
 }
 
+final class EitherIdOps[A](val obj: A) extends AnyVal {
```

Comment:
You will also have to fix `ApplicativeIdOps`, `ValidatedIdSyntax`, and `WriterIdSyntax`. I don't think e.g. attaching an `oa` member to every `Option[A]` is particularly helpful either; are you sure we shouldn't make them all private?